02-intermediate-front-end/events-and-callbacks/readme.md

```diff
@@ -33,77 +76,592 @@ button.addEventListener("click", handleClickEvent)
+
+## Timing functions
+
+Let's look at timing functions -- that is, Javascript's way of making something happen every `x` seconds.
+
+Replace the contents of your `script.js` with this:
+
+```js
+function sayHello(){
+  console.log("Hi there!")
+}
+setInterval(sayHello, 1000);
+```
+
+### Turn and talk
+
+- What does the number in `setInterval` indicate?
+- Replace `setInterval` with `setTimeout`. What's the difference?
+
+We'll make it more interesting by having the timer start on a click event, and stop on another click event.
+
+Put a "start" and a "stop" button in your HTML:
+
+```html
+<button id="start">Start</button>
+<button id="stop">Stop</button>
+```
+
+Then, replace the contents of your `script.js` with this:
+
+```js
+var start = document.getElementById("start");
+var stop = document.getElementById("stop");
+var singAnnoyingSong = function(){
+  console.log("I know a song that gets on everybody's nerves...")
+}
+var songTimer;
+start.addEventListener("click", function(){
+  songTimer = setInterval(singAnnoyingSong, 100);
+});
+stop.addEventListener("click", function(){
+  clearInterval(songTimer);
+});
+```
+
+### Turn and talk
+
+- What happens when you click the "start" button a bunch of times in a row?
+  - Why?
+  - How is this different from events?
+  - When you do this, why doesn't the "stop" button seem to work?
+- How is `clearInterval` different from `removeEventListener`?
+- Give `singAnnoyingSong` an argument of `e`, like we did for the event listeners. What information does it contain?
+
+## Making your own callbacks
+
+We've done a lot of stuff with callbacks today, passing them as arguments into built-in Javascript functions. But we haven't tried passing them into functions we wrote ourselves.
+
+We're going to create a script that shows your interal monologue over the course of a normal weekday. Create three functions: one that `console.logs` something you think to yourself on waking up, another something you think during your commute, and that last something you think when you get to GA.
+
+Then, create a function called `averageMorning` that takes three arguments, `first`, `then`, and `lastly`.
+
+These three arguments represent three functions. Inside `averageMorning`, call `first`, `then`, and `lastly`, in order.
+
+Then, call the `averageMorning` function itself, passing in as arguments the three functions you wrote.
+
+The result should be something like this:
+
+```js
+function haveBreakfast(){
+  console.log("Gotta have my Pops!")
+}
+function commute(){
+  console.log("What an interesting smell on Metro this morning...")
+}
+function workAtGA(){
+  console.log("At work! Now I can promote RobertAKARobin.com through the disguise of teaching again.")
+}
+function averageMorning(first, then, lastly){
+  console.log("This is my internal monologue on a normal weekday")
+  first();
+  then();
+  setTimeout(lastly, 5000);
+}
+averageMorning(haveBreakfast, commute, workAtGA);
+```
+
+## IIFEs
```

Comment:
I dunno. I think this is necessary, because otherwise I don't think they get why sometimes you just write `functionName` and other times you write `functionName()`.